### PR TITLE
add redux

### PR DESCRIPTION
**Pull request by avishag-gvili**

### CODE REVIEW

**Review by Shoshana-Yudelevitz — COMMENTED** [review]
הוספת REDUX לפרויקט

File: client-side/src/redux/auth/auth.slice.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    user: null,
+    isAuthenticated: false,
+    isInitialized: false
+}
+
+const authSlice = createSlice({
+    name: 'auth',
+    initialState,
+    reducers:{
+        setCurentUser: (state: AuthStateType, action: PayloadAction<AuthUser>) => {
```
Comment (SariGlick) [inline client-side/src/redux/auth/auth.slice.ts:20]: fix curent to current

File: client-side/src/redux/auth/auth.slice.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            state.user = action.payload;
+            state.isAuthenticated = true;
+            state.isInitialized = true;
+        },
+        setInitialized: (state: AuthStateType) => {
+            state.isInitialized = true
+        },
+        deleteCurentUser:(state: AuthStateType) => {
```
Comment (SariGlick) [inline client-side/src/redux/auth/auth.slice.ts:28]: fix curent to current

File: client-side/src/redux/auth/auth.slice.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            state.user = null;
+            state.isAuthenticated = false;
+            state.isInitialized = false;
+        },
+        updateCurentUser: (state: AuthStateType, action: PayloadAction<AuthUser>) => {
```
Comment (SariGlick) [inline client-side/src/redux/auth/auth.slice.ts:33]: fix curent to current

File: client-side/src/redux/preference/preference.slice.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    preferences: Preference[]
+}
+
+const initialState: preferenceStateType = { preferences:[] }
+
+const preferenceSlice = createSlice({
+    name: 'preference',
+    initialState,
+    reducers: {
+        setpreferenceSlice: (state, action: PayloadAction<Preference[]>) => {
```
Comment (SariGlick) [inline client-side/src/redux/preference/preference.slice.ts:14]: replace p with upper case

File: client-side/src/redux/preference/preference.slice.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+            state.preferences = action.payload;
+        },
+        addpreferenceSlice: (state, action: PayloadAction<Preference>) => {
```
Comment (SariGlick) [inline client-side/src/redux/preference/preference.slice.ts:17]: replace p with upper case

File: client-side/src/redux/preference/preference.slice.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+            state.preferences.push(action.payload)
+        },
+        updatepreferenceSlice: (state, action: PayloadAction<Preference>) => {
```
Comment (SariGlick) [inline client-side/src/redux/preference/preference.slice.ts:20]: replace p with upper case

File: client-side/src/redux/preference/preference.slice.ts [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+            const index = state.preferences.findIndex(profile => profile.id === action.payload.id);
+            if (index !== -1) {
+                state.preferences[index] = action.payload;
+            }
+        },
+        deletepreferenceSlice: (state, action: PayloadAction<string>) => {
```
Comment (SariGlick) [inline client-side/src/redux/preference/preference.slice.ts:26]: replace p with upper case

File: client-side/src/redux/profile/profile.slice.ts [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+const initialState: profileStateType = { profiles:[] }
+
+const profileSlice = createSlice({
+    name: 'profile',
+    initialState,
+    reducers: {
+        setProfile: (state, action: PayloadAction<Profile[]>) => {
+            state.profiles = action.payload;
+        },
+        addprofile: (state, action: PayloadAction<Profile>) => {
```
Comment (SariGlick) [inline client-side/src/redux/profile/profile.slice.ts:17]: replace p with upper case

File: client-side/src/redux/profile/profile.slice.ts [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+            state.profiles.push(action.payload)
+        },
+        updateprofile: (state, action: PayloadAction<Profile>) => {
```
Comment (SariGlick) [inline client-side/src/redux/profile/profile.slice.ts:20]: replace p with upper case

File: client-side/src/redux/profile/profile.slice.ts [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+            const index = state.profiles.findIndex(profile => profile.id === action.payload.id);
+            if (index !== -1) {
+                state.profiles[index] = action.payload;
+            }
+        },
+        deleteprofile: (state, action: PayloadAction<string>) => {
```
Comment (SariGlick) [inline client-side/src/redux/profile/profile.slice.ts:26]: replace p with upper case

File: client-side/src/redux/visitedWebsites/visitedWebsites.slice.ts [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+    visitedWebsites: VisitedWebsites[]
+}
+
+const initialState: visitedWebsitesStateType = { visitedWebsites:[] }
+
+const visitedWebsiteSlice = createSlice({
+    name: 'profile',
+    initialState,
+    reducers: {
+        setvisitedWebsite: (state, action: PayloadAction<VisitedWebsites[]>) => {
```
Comment (SariGlick) [inline client-side/src/redux/visitedWebsites/visitedWebsites.slice.ts:14]: replace v with upper case

File: client-side/src/redux/visitedWebsites/visitedWebsites.slice.ts [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+            state.visitedWebsites = action.payload;
+        },
+        addvisitedWebsite: (state, action: PayloadAction<VisitedWebsites>) => {
```
Comment (SariGlick) [inline client-side/src/redux/visitedWebsites/visitedWebsites.slice.ts:17]: replace v with upper case

File: client-side/src/redux/visitedWebsites/visitedWebsites.slice.ts [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+            state.visitedWebsites.push(action.payload)
+        },
+        updatevisitedWebsite: (state, action: PayloadAction<VisitedWebsites>) => {
```
Comment (SariGlick) [inline client-side/src/redux/visitedWebsites/visitedWebsites.slice.ts:20]: replace v with upper case

File: client-side/src/redux/visitedWebsites/visitedWebsites.slice.ts [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+            const index = state.visitedWebsites.findIndex(visitedWebsite => visitedWebsite.id === action.payload.id);
+            if (index !== -1) {
+                state.visitedWebsites[index] = action.payload;
+            }
+        },
+        deletevisitedWebsite: (state, action: PayloadAction<string>) => {
```
Comment (SariGlick) [inline client-side/src/redux/visitedWebsites/visitedWebsites.slice.ts:26]: replace v with upper case

File: client-side/src/services/profileService.ts [hunk 15]
```diff
@@ -0,0 +1,60 @@
...
+        const response = await axios.post('/profiles', profileData);
+        return response.data;
+    } catch (err) {
+        console.error('Error creating profile:', err);
+        throw err;
+    }
+};
+export const getProfileById = async (id) => {
+    try {
+        console.log(id,"id");
```
Comment (SariGlick) [inline client-side/src/services/profileService.ts:23]: remove all unnecessary console.log

**Review by SariGlick — CHANGES_REQUESTED** [review]

**Pull request closed without merging**